### PR TITLE
feat: setup commit lint and semantic release

### PR DESCRIPTION
**Pull request by umglurf**

### CODE REVIEW

**Comment by umglurf** [comment]
Tested here: https://github.com/nrkno/iac-test-release

File: .github/workflows/workflow.yaml [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+          else
+            echo "Using workflow configuration for commitlint"
+            echo "::set-output name=commitlintconfig::.github-workflow-semantic-release/commitlint.config.js"
+          fi
+      - uses: wagoid/commitlint-github-action@v4
+        id: lint
+        continue-on-error: ${{ github.event_name == 'pull_request' }}
+        with:
+          configFile: ${{ steps.set_config.outputs.commitlintconfig }}
+      - uses: actions/github-script@v6
```
Comment (stigok) [inline .github/workflows/workflow.yaml:62]: Might be runner-specific whether we have Node.js installed or not. Could be smart to add an `setup-nodejs` job here.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,36 @@
-# github-workflow-semantic-release
-Semantic Release workflow through github action
+# Semantic Release workflow through github action
```
Comment (stigok) [inline README.md:1]: GitHub Actions

File: README.md [hunk 3]
```diff
@@ -1,2 +1,36 @@
...
+Reusable Github workflow for linting commits and make releases using
+[semantic-release](https://github.com/semantic-release/semantic-release/)
+
+## Usage
+
+This workflow will use sensible defaults for
+[commintlint](https://commitlint.js.org) and
+[semantic-release](https://github.com/semantic-release/semantic-release/)
+
+If you want to ovveride the
```
Comment (stigok) [inline README.md:12]: override

**Review by stigok — CHANGES_REQUESTED** [review]

File: .github/workflows/lint-release.yaml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+---
+on: [push, pull_request]
+
+name: Commit lint and release
+jobs:
+  lint_release:
+    uses: nrkno/github-workflow-semantic-release/.github/workflows/workflow.yaml@v1
```
[the remark this answers is not in the document]
Comment (umglurf) [reply .github/workflows/lint-release.yaml:7]: If we point to main, we loose the whole point of versioning. If there is an error, we need to quickly release a fix instead, moving forward.

File: .github/workflows/workflow.yaml (hunk 1 above)
Comment (umglurf) [reply .github/workflows/workflow.yaml:62]: That is the responsibility of that action to install deps, not us.

File: .github/workflows/lint-release.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (umglurf) [reply .github/workflows/lint-release.yaml:7]: changed

File: README.md (hunk 3 above)
Comment (umglurf) [reply README.md]: fixe

File: README.md (hunk 2 above)
Comment (umglurf) [reply README.md]: fixed

**Review by stigok — APPROVED** [review]

**Pull request merged**